client/mm/libxc/binance.go

```diff
@@ -0,0 +1,1190 @@
+		if binQty >= remaining {
+			filled = true
+			weightedSum += remaining * extrema
+			break
+		}
+		remaining -= binQty
+		weightedSum += binQty * extrema
+	}
+
+	if !filled {
+		return 0, 0, false, nil
+	}
+
+	return weightedSum / qty, extrema, true, nil
+}
+
+type binanceNetworkInfo struct {
+	AddressRegex            string  `json:"addressRegex"`
+	Coin                    string  `json:"coin"`
+	DepositEnable           bool    `json:"depositEnable"`
+	IsDefault               bool    `json:"isDefault"`
+	MemoRegex               string  `json:"memoRegex"`
+	MinConfirm              int     `json:"minConfirm"`
+	Name                    string  `json:"name"`
+	Network                 string  `json:"network"`
+	ResetAddressStatus      bool    `json:"resetAddressStatus"`
+	SpecialTips             string  `json:"specialTips"`
+	UnLockConfirm           int     `json:"unLockConfirm"`
+	WithdrawEnable          bool    `json:"withdrawEnable"`
+	WithdrawFee             float64 `json:"withdrawFee,string"`
+	WithdrawIntegerMultiple float64 `json:"withdrawIntegerMultiple,string"`
+	WithdrawMax             float64 `json:"withdrawMax,string"`
+	WithdrawMin             float64 `json:"withdrawMin,string"`
+	SameAddress             bool    `json:"sameAddress"`
+	EstimatedArrivalTime    int     `json:"estimatedArrivalTime"`
+	Busy                    bool    `json:"busy"`
+}
+
+type binanceCoinInfo struct {
+	Coin              string                `json:"coin"`
+	DepositAllEnable  bool                  `json:"depositAllEnable"`
+	Free              float64               `json:"free,string"`
+	Freeze            float64               `json:"freeze,string"`
+	Ipoable           float64               `json:"ipoable,string"`
+	Ipoing            float64               `json:"ipoing,string"`
+	IsLegalMoney      bool                  `json:"isLegalMoney"`
+	Locked            float64               `json:"locked,string"`
+	Name              string                `json:"name"`
+	Storage           float64               `json:"storage,string"`
+	Trading           bool                  `json:"trading"`
+	WithdrawAllEnable bool                  `json:"withdrawAllEnable"`
+	Withdrawing       float64               `json:"withdrawing,string"`
+	NetworkList       []*binanceNetworkInfo `json:"networkList"`
+}
+
+type symbol struct {
+	Symbol              string   `json:"symbol"`
+	Status              string   `json:"status"`
+	BaseAsset           string   `json:"baseAsset"`
+	BaseAssetPrecision  int      `json:"baseAssetPrecision"`
+	QuoteAsset          string   `json:"quoteAsset"`
+	QuoteAssetPrecision int      `json:"quoteAssetPrecision"`
+	OrderTypes          []string `json:"orderTypes"`
+}
+
+// markets returns all the possible markets for this symbol. A symbol
+// represents a single market on the CEX, but tokens on the DEX have a
+// different assetID for each network they are on, therefore they will
+// match multiple markets as defined using assetID.
```

Comment:
> The CEX will only allow deposits and withdraws on probably 1 chain.

This is not true. You can deposit USDC from many different chains.

Reply:
If you offered me to arbitrage with a token that I had some of on say Polygon, and then later I found out I couldn't withdraw on Polygon and would have to use a bridge and pay fee on Ethereum, I'd be pretty upset. I don't see a scenario where we should be trading tokens that we can't withdraw with our own wallet.

Reply:
Acutally, that is the case with Polygon on mainnet right now.
```
{
    "coin": "USDC",
    ...
    "networkList": [
        {

            "depositEnable": true,
            "name": "Polygon",
            "network": "MATIC",
            "withdrawEnable": false,
        }, ...
    ]
}
```
I don't know if withdraws to Polygon are ever enabled or if this is temporary.

Reply:
We should put a warning in that case, but I don't think it should be blocked. I was even thinking about adding a feature for a "similar asset" setting, in which I could set USDT as a similar asset to USDC, and then arb that way. If the DCR/USDC(polygon) market gets used, but Binance only has a DCR/USDT market, wouldn't it be good to be able to arb them? These would be the only two not completely spoofed DCR/stablecoin markets.

Reply:
Maybe others can weigh in, but the idea of trading something on a DEX that I can't immediately withdraw is crazy to me.

I'm trying to see the utility of trading something I can't withdraw to my wallet. I guess I always thought that a profitable arbitrage bot would just require the ability to transfer funds between markets to avoid getting jammed, but I don't have any proof of that theory.

Reply:
> I guess I always thought that a profitable arbitrage bot would just require the ability to transfer funds between markets to avoid getting jammed,

You mean automatically rebalancing between the CEX and DEX balances? That's definitely ideal.. but if I have a couple lots worth of funds sitting around, I could stick half on the CEX, half on the DEX and start the bot. If the balance of one of the assets gets too low, then yes the bot will stop working until I manually rebalance it. I could bridge, or transfer to another CEX that allows withdrawals on the chain I want. I don't see why this should be blocked, as long as it's very clear for the user what's going to happen.

Thoughts? @JoeGruffins @chappjc